components/org.wso2.carbon.identity.oauth.uma.resource.service/src/main/java/org/wso2/carbon/identity/oauth/uma/resource/service/dao/ResourceDAO.java

```diff
@@ -191,15 +182,14 @@ public static Resource retrieveResource(String resourceid) throws UMAServiceExce
                     if (resultSet.getString("PROPERTY_KEY").equals(DESCRIPTION)) {
                         resourceRegistration.setDescription(resultSet.getString("PROPERTY_VALUE"));
                     }
-                }
+                } while (resultSet.next());
+            }
+            if (log.isDebugEnabled()) {
+                log.debug("Successfully retrieved the resource details from the database.");
             }
-            log.info("Successfully retrieved the resource details from the database.");
         } catch (SQLException e) {
-            log.error("Error when retrieving the resource description. ");
-            throw new UMAServiceException(ResourceConstants.ErrorMessages.ERROR_CODE_NOT_FOUND_RESOURCE_ID,
-                    "Database" +
-                            "error.Could not get resource.Resource Id can not be found in data base. - " +
-                            e.getMessage(), e);
+            throw new UMAServiceException(ResourceConstants.ErrorMessages.ERROR_CODE_NOT_FOUND_RESOURCE_ID, "Database" +
+                    "error.Could not get resource.Resource Id can not be found in data base. - ", e);
```

Comment:
remove trailing '-'
Include context information (e.g. resource id) in the error message.